corehq/apps/userreports/tests/test_choice_provider.py

```diff
@@ -133,49 +137,80 @@ def test_get_choices_for_values(self):
-            ('cambridge_ma', 'Cambridge'),
-            ('somerville_ma', 'Somerville'),
-            ('boston_ma', 'Boston'),
-        )
-        cls.locations = []
-        choices = []
-
-        for location_code, location_name in location_code_name_pairs:
-            location = cls.make_location(location_code, location_name)
-            cls.locations.append(location)
-            choices.append(SearchableChoice(location.location_id, location.sql_location.display_name,
-                                            searchable_text=[location_code, location_name]))
+        choices = [
+            SearchableChoice(
+                location.location_id,
+                location.display_name,
+                searchable_text=[location.site_code, location.name]
+            )
+            for location in cls.locations.itervalues()
+        ]
         choices.sort(key=lambda choice: choice.display)
+        cls.web_user = WebUser.create(cls.domain, 'blah', 'password')
         cls.choice_provider = LocationChoiceProvider(report, None)
         cls.static_choice_provider = StaticChoiceProvider(choices)
+        cls.choice_query_context = partial(ChoiceQueryContext, user=cls.web_user)
 
     @classmethod
     def tearDownClass(cls):
         cls.domain_obj.delete()
         delete_all_locations()
 
     def test_query_search(self):
-        self._test_query(ChoiceQueryContext('e', 2, page=0))
-        self._test_query(ChoiceQueryContext('e', 2, page=1))
+        self._test_query(self.choice_query_context('e', page=0))
+        self._test_query(self.choice_query_context('e', page=1))
 
     def test_get_choices_for_values(self):
         self._test_get_choices_for_values(
-            ['made-up', self.locations[0].location_id, self.locations[1].location_id])
+            ['made-up', self.locations['Cambridge'].location_id, self.locations['Middlesex'].location_id])
+
+    def test_scoped_to_location_search(self):
+        self.web_user.set_location(self.domain, self.locations['Middlesex'])
+        role = UserRole(
+            domain=self.domain,
+            name='Regional Supervisor',
+            permissions=Permissions(access_all_locations=False),
+        )
+        role.save()
+        self.web_user.set_role(self.domain, role.get_qualified_id())
+
+        scoped_choices = [
+            SearchableChoice(
+                location.location_id,
+                location.display_name,
+                searchable_text=[location.site_code, location.name]
+            )
+            for location in [
+                self.locations['Cambridge'],
+                self.locations['Middlesex'],
+                self.locations['Somerville'],
+            ]
+        ]
+        self.static_choice_provider = StaticChoiceProvider(scoped_choices)
+        self._test_query(self.choice_query_context('', page=0))
+        self._test_query(self.choice_query_context('Somerville', page=0))
+        self._test_query(self.choice_query_context('Boston', page=0))
```

Comment:
These tests are hard to follow, but it sounds right.  Essentially they're comparing that `self.static_choice_provider` and `self.choice_provider` return the same results for the provided queries.